# Analyzing of Used Vehicles Posted on the German eBay Website

## Dataset introduction

There are 50,000 rows in the dataset. That means there are 50,000 ads. The dataset gives us information such as when the cars were published on the site, their price, and the last date they were viewed. The columns of the dataset are as follows:

- dateCrawled - When this ad was first crawled. All field-values are taken from this date.
- name - Name of the car.
- seller - Whether the seller is private or a dealer.
- offerType - The type of listing
- price - The price on the ad to sell the car.
- abtest - Whether the listing is included in an A/B test.
- vehicleType - The vehicle Type.
- yearOfRegistration - The year in which the car was first registered.
- gearbox - The transmission type.
- powerPS - The power of the car in PS.
- model - The car model name.
- kilometer - How many kilometers the car has driven.
- monthOfRegistration - The month in which the car was first registered.
- fuelType - What type of fuel the car uses.
- brand - The brand of the car.
- notRepairedDamage - If the car has a damage which is not yet repaired.
- dateCreated - The date on which the eBay listing was created.
- nrOfPictures - The number of pictures in the ad.
- postalCode - The postal code for the location of the vehicle.
- lastSeenOnline - When the crawler saw this ad last online.

###### The aim of the project is to clean up the dataset and analyze the included used car lists. 
Now that we know what information the columns in the data set contain, let's take a closer look at the column contents.

In [1]:
import pandas as pd
import numpy as np

In [2]:
autos = pd.read_csv('autos.csv',encoding='Latin-1')

In [3]:
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

       seller offerType    price   abtest vehicleType  yearOfRegistration  \
0      privat   Angebot   $5,000  control         bus                2004   
1      privat   Angebot   $8,500  control   limousine                1997   
2      privat   Angebot   $8,990     test   limousine                2009   
3      privat   Angebot   $4,350  control  kleinwagen                2007   
4      privat   Angebot   $1,350     test       kombi                2003   
...       ...       ...      ...      ...         ...                 ...   
49995  privat   Angebot  $24,900  control   limousine                2011   
49996  privat   Angebot   $1,980  control      cabrio                1996   
49997  privat   Angebot  $13,200     test      cabrio                2014   
49998  privat   Angebot  $22,900  control       kombi                2013   
49999  privat   Angebot   $1,250  control   limousine                1996   

         gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0        manuell      158  andere  150,000km                    3      lpg   
1      automatik      286     7er  150,000km                    6   benzin   
2        manuell      102    golf   70,000km                    7   benzin   
3      automatik       71  fortwo   70,000km                    6   benzin   
4        manuell        0   focus  150,000km                    7   benzin   
...          ...      ...     ...        ...                  ...      ...   
49995  automatik      239      q5  100,000km                    1   diesel   
49996    manuell       75   astra  150,000km                    5   benzin   
49997  automatik       69     500    5,000km                   11   benzin   
49998    manuell      150      a3   40,000km                   11   diesel   
49999    manuell      101  vectra  150,000km                    1   benzin   

            brand notRepairedDamage          dateCreated  nrOfPictures  \
0         peugeot              nein  2016-03-26 00:00:00             0   
1             bmw              nein  2016-04-04 00:00:00             0   
2      volkswagen              nein  2016-03-26 00:00:00             0   
3           smart              nein  2016-03-12 00:00:00             0   
4            ford              nein  2016-04-01 00:00:00             0   
...           ...               ...                  ...           ...   
49995        audi              nein  2016-03-27 00:00:00             0   
49996        opel              nein  2016-03-28 00:00:00             0   
49997        fiat              nein  2016-04-02 00:00:00             0   
49998        audi              nein  2016-03-08 00:00:00             0   
49999        opel              nein  2016-03-13 00:00:00             0   

       postalCode             lastSeen  
0           79588  2016-04-06 06:45:54  
1           71034  2016-04-06 14:45:08  
2           35394  2016-04-06 20:15:37  
3           33729  2016-03-15 03:16:28  
4           39218  2016-04-01 14:38:50  
...           ...                  ...  
4999

In [4]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   dateCrawled          50000 non-null  object
 1   name                 50000 non-null  object
 2   seller               50000 non-null  object
 3   offerType            50000 non-null  object
 4   price                50000 non-null  object
 5   abtest               50000 non-null  object
 6   vehicleType          44905 non-null  object
 7   yearOfRegistration   50000 non-null  int64 
 8   gearbox              47320 non-null  object
 9   powerPS              50000 non-null  int64 
 10  model                47242 non-null  object
 11  odometer             50000 non-null  object
 12  monthOfRegistration  50000 non-null  int64 
 13  fuelType             45518 non-null  object
 14  brand                50000 non-null  object
 15  notRepairedDamage    40171 non-null  object
 16  date

In [5]:
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

The data set consists of 50 thousand rows and 20 columns. There are empty values in 5 columns (vehicleType, gearbox, model, fuelType, notRepairedDamage). The data type of the 5 columns is int64(yearOfRegistration, powerPS, monthOfRegistration, nrOfPictures, postalCode). The remaining data types are object.

In [6]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [7]:
new_columns = []
def correcting_col(col_name,new_name):
    new_col_name = col_name.replace(col_name,new_name)
    new_columns.append(new_col_name)

correcting_col('dateCrawled','date_crawled')
correcting_col('name','name')
correcting_col('seller','seller')
correcting_col('offerType','offer_type')
correcting_col('price','price')
correcting_col('abtest','abtest')
correcting_col('vehicleType','vehicle_type')
correcting_col('yearOfRegistration','registration_year')
correcting_col('gearbox','gear_box')
correcting_col('powerPS','power_ps')
correcting_col('model','model')
correcting_col('odometer','odometer')
correcting_col('monthOfRegistration','registration_month')
correcting_col('fuelType','fuel_type')
correcting_col('brand','brand')
correcting_col('notRepairedDamage','unrepaired_damage')
correcting_col('dateCreated','ad_created')
correcting_col('nrOfPictures','nr_pictures')
correcting_col('postalCode','postal_code')
correcting_col('lastSeen','last_seen')

In [8]:
autos.columns = new_columns

In [9]:
autos.columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'abtest',
       'vehicle_type', 'registration_year', 'gear_box', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damage', 'ad_created', 'nr_pictures', 'postal_code',
       'last_seen'],
      dtype='object')

In [10]:
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

    gear_box  power_ps   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  nr_pictures  \
0     peugeot              nein  2016-03-26 00:00:00            0   
1         bmw              nein  2016-04-04 00:00:00            0   
2  volkswagen              nein  2016-03-26 00:00:00            0   
3       smart              nein  2016-03-12 00:00:00            0   
4        ford              nein  2016-04-01 00:00:00            0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

Some columns were written in camelcase structure. So we've converted them to the snakecase structure, which is more readable. This way it will be easier for us to analyze the data.

In [11]:
autos.describe(include='all').T

count unique                  top   freq     mean  \
date_crawled        50000  48213  2016-03-12 16:06:22      3      NaN   
name                50000  38754          Ford_Fiesta     78      NaN   
seller              50000      2               privat  49999      NaN   
offer_type          50000      2              Angebot  49999      NaN   
price               50000   2357                   $0   1421      NaN   
abtest              50000      2                 test  25756      NaN   
vehicle_type        44905      8            limousine  12859      NaN   
registration_year   50000    NaN                  NaN    NaN  2005.07   
gear_box            47320      2              manuell  36993      NaN   
power_ps            50000    NaN                  NaN    NaN  116.356   
model               47242    245                 golf   4024      NaN   
odometer            50000     13            150,000km  32424      NaN   
registration_month  50000    NaN                  NaN    NaN  5.72336   
fuel_type           45518      7               benzin  30107      NaN   
brand               50000     40           volkswagen  10687      NaN   
unrepaired_damage   40171      2                 nein  35232      NaN   
ad_created          50000     76  2016-04-03 00:00:00   1946      NaN   
nr_pictures         50000    NaN                  NaN    NaN        0   
postal_code         50000    NaN                  NaN    NaN  50813.6   
last_seen           50000  39481  2016-04-07 06:17:27      8      NaN   

                        std   min    25%    50%    75%    max  
date_crawled            NaN   NaN    NaN    NaN    NaN    NaN  
name                    NaN   NaN    NaN    NaN    NaN    NaN  
seller                  NaN   NaN    NaN    NaN    NaN    NaN  
offer_type              NaN   NaN    NaN    NaN    NaN    NaN  
price                   NaN   NaN    NaN    NaN    NaN    NaN  
abtest                  NaN   NaN    NaN    NaN    NaN    NaN  
vehicle_type            NaN   NaN    NaN    NaN    NaN    NaN  
registration_year   105.713  1000   1999   2003   2008   9999  
gear_box                NaN   NaN    NaN    NaN    NaN    NaN  
power_ps            209.217     0     70    105    150  17700  
model                   NaN   NaN    NaN    NaN    NaN    NaN  
odometer                NaN   NaN    NaN    NaN    NaN    NaN  
registration_month  3.71198     0      3      6      9     12  
fuel_type               NaN   NaN    NaN    NaN    NaN    NaN  
brand                   NaN   NaN    NaN    NaN    NaN    NaN  
unrepaired_damage       NaN   NaN    NaN    NaN    NaN    NaN  
ad_created              NaN   NaN    NaN    NaN    NaN    NaN  
nr_pictures               0     0      0      0      0      0  
postal_code         25779.7  1067  30451  49577  71540  99998  
last_seen               NaN   NaN    NaN    NaN    NaN    NaN

Some columns have data with only two frequencies, but almost all of them have the same value (seller, offer_type). Some columns do not contain useful data (postal_code, nr_pictures).

In [12]:
autos['vehicle_type'].value_counts()

limousine     12859
kleinwagen    10822
kombi          9127
bus            4093
cabrio         3061
coupe          2537
suv            1986
andere          420
Name: vehicle_type, dtype: int64

In [13]:
autos['fuel_type'].value_counts()

benzin     30107
diesel     14567
lpg          691
cng           75
hybrid        37
andere        22
elektro       19
Name: fuel_type, dtype: int64

In [14]:
autos['gear_box'].value_counts()

manuell      36993
automatik    10327
Name: gear_box, dtype: int64

The most published vehicle features are:
- ###### type limousine
- ###### fuel type gasoline
- ###### gearbox manual

In [15]:
autos['price'] = autos['price'].str.replace(',','').str.replace('$','').astype(int)

In [16]:
autos['odometer'] = autos['odometer'].str.replace(',','').str.replace('km','').astype(int)

In [17]:
autos.rename(columns={'odometer':'odometer_km'},inplace=True)

In [18]:
autos['price'].unique().shape

(2357,)

In [19]:
autos['price'].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [20]:
autos['price'].value_counts().sort_index(ascending=True).head(60)

0      1421
1       156
2         3
3         1
5         2
8         1
9         1
10        7
11        2
12        3
13        2
14        1
15        2
17        3
18        1
20        4
25        5
29        1
30        7
35        1
40        6
45        4
47        1
49        4
50       49
55        2
59        1
60        9
65        5
66        1
70       10
75        5
79        1
80       15
89        1
90        5
99       19
100     134
110       3
111       2
115       2
117       1
120      39
122       1
125       8
129       1
130      15
135       1
139       1
140       9
145       2
149       7
150     224
156       2
160       8
170       7
173       1
175      12
179       1
180      35
Name: price, dtype: int64

In [21]:
autos['price'].value_counts().sort_index(ascending=True).tail(60)

84000       1
84997       1
85000       1
86500       1
88900       1
89000       1
89900       1
93000       2
93911       1
94999       1
98500       1
99000       2
99900       2
104900      1
105000      2
109999      1
114400      1
115000      1
115991      1
116000      1
119500      1
119900      1
120000      2
128000      1
129000      1
130000      1
135000      1
137999      1
139997      1
145000      1
151990      1
155000      1
163500      1
163991      1
169000      1
169999      1
175000      1
180000      1
190000      1
194000      1
197000      1
198000      1
220000      1
250000      1
259000      1
265000      1
295000      1
299000      1
345000      1
350000      1
999990      1
999999      2
1234566     1
1300000     1
3890000     1
10000000    1
11111111    2
12345678    3
27322222    1
99999999    1
Name: price, dtype: int64

We will delete cars with a price of 0. Because this is false data. We will also delete those with a price greater than 350,000. Because they are outliers.

In [22]:
autos = autos[(autos['price'] >= 1) & (autos['price'] <= 350000)]

In [23]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 48565 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   date_crawled        48565 non-null  object
 1   name                48565 non-null  object
 2   seller              48565 non-null  object
 3   offer_type          48565 non-null  object
 4   price               48565 non-null  int64 
 5   abtest              48565 non-null  object
 6   vehicle_type        43979 non-null  object
 7   registration_year   48565 non-null  int64 
 8   gear_box            46222 non-null  object
 9   power_ps            48565 non-null  int64 
 10  model               46107 non-null  object
 11  odometer_km         48565 non-null  int64 
 12  registration_month  48565 non-null  int64 
 13  fuel_type           44535 non-null  object
 14  brand               48565 non-null  object
 15  unrepaired_damage   39464 non-null  object
 16  ad_created          48

Initially there were 50,000 rows of data. The number of vehicles remaining after deleting outlier data is 48565.

In [24]:
autos['odometer_km'].value_counts(ascending=True)

10000       253
20000       762
30000       780
40000       815
5000        836
50000      1012
60000      1155
70000      1217
80000      1415
90000      1734
100000     2115
125000     5057
150000    31414
Name: odometer_km, dtype: int64

The values in the odometer_km column look normal.

In [25]:
autos['date_crawled'].str[:10].value_counts(normalize=True,dropna=False).sort_index(ascending=True)

2016-03-05    0.025327
2016-03-06    0.014043
2016-03-07    0.036014
2016-03-08    0.033296
2016-03-09    0.033090
2016-03-10    0.032184
2016-03-11    0.032575
2016-03-12    0.036920
2016-03-13    0.015670
2016-03-14    0.036549
2016-03-15    0.034284
2016-03-16    0.029610
2016-03-17    0.031628
2016-03-18    0.012911
2016-03-19    0.034778
2016-03-20    0.037887
2016-03-21    0.037373
2016-03-22    0.032987
2016-03-23    0.032225
2016-03-24    0.029342
2016-03-25    0.031607
2016-03-26    0.032204
2016-03-27    0.031092
2016-03-28    0.034860
2016-03-29    0.034099
2016-03-30    0.033687
2016-03-31    0.031834
2016-04-01    0.033687
2016-04-02    0.035478
2016-04-03    0.038608
2016-04-04    0.036487
2016-04-05    0.013096
2016-04-06    0.003171
2016-04-07    0.001400
Name: date_crawled, dtype: float64

In [26]:
autos['last_seen'].str[:10].value_counts(normalize=True,dropna=False).sort_index(ascending=True)

2016-03-05    0.001071
2016-03-06    0.004324
2016-03-07    0.005395
2016-03-08    0.007413
2016-03-09    0.009595
2016-03-10    0.010666
2016-03-11    0.012375
2016-03-12    0.023783
2016-03-13    0.008895
2016-03-14    0.012602
2016-03-15    0.015876
2016-03-16    0.016452
2016-03-17    0.028086
2016-03-18    0.007351
2016-03-19    0.015834
2016-03-20    0.020653
2016-03-21    0.020632
2016-03-22    0.021373
2016-03-23    0.018532
2016-03-24    0.019767
2016-03-25    0.019211
2016-03-26    0.016802
2016-03-27    0.015649
2016-03-28    0.020859
2016-03-29    0.022341
2016-03-30    0.024771
2016-03-31    0.023783
2016-04-01    0.022794
2016-04-02    0.024915
2016-04-03    0.025203
2016-04-04    0.024483
2016-04-05    0.124761
2016-04-06    0.221806
2016-04-07    0.131947
Name: last_seen, dtype: float64

In [27]:
autos['ad_created'].str[:10].value_counts(normalize=True,dropna=False).sort_index().head(60)

2015-06-11    0.000021
2015-08-10    0.000021
2015-09-09    0.000021
2015-11-10    0.000021
2015-12-05    0.000021
2015-12-30    0.000021
2016-01-03    0.000021
2016-01-07    0.000021
2016-01-10    0.000041
2016-01-13    0.000021
2016-01-14    0.000021
2016-01-16    0.000021
2016-01-22    0.000021
2016-01-27    0.000062
2016-01-29    0.000021
2016-02-01    0.000021
2016-02-02    0.000041
2016-02-05    0.000041
2016-02-07    0.000021
2016-02-08    0.000021
2016-02-09    0.000021
2016-02-11    0.000021
2016-02-12    0.000041
2016-02-14    0.000041
2016-02-16    0.000021
2016-02-17    0.000021
2016-02-18    0.000041
2016-02-19    0.000062
2016-02-20    0.000041
2016-02-21    0.000062
2016-02-22    0.000021
2016-02-23    0.000082
2016-02-24    0.000041
2016-02-25    0.000062
2016-02-26    0.000041
2016-02-27    0.000124
2016-02-28    0.000206
2016-02-29    0.000165
2016-03-01    0.000103
2016-03-02    0.000103
2016-03-03    0.000865
2016-03-04    0.001483
2016-03-05    0.022897
2016-03-06 

In [28]:
autos['ad_created'].str[:10].value_counts(normalize=True,dropna=False).sort_index().tail(16)

2016-03-23    0.032060
2016-03-24    0.029280
2016-03-25    0.031751
2016-03-26    0.032266
2016-03-27    0.030989
2016-03-28    0.034984
2016-03-29    0.034037
2016-03-30    0.033501
2016-03-31    0.031875
2016-04-01    0.033687
2016-04-02    0.035149
2016-04-03    0.038855
2016-04-04    0.036858
2016-04-05    0.011819
2016-04-06    0.003253
2016-04-07    0.001256
Name: ad_created, dtype: float64

Ads started to be created in 2015, but there is very limited data at that time. Advertisements were posted almost every day in February 2016. Advertisement frequencies seem to be the same, but increased towards March. In March, advertisements are posted every day and usually appear at the same frequency values. In the last 3 lines of the data for April, there was a decrease in advertisements. Ads were first scanned on March 5, 2016.

In [29]:
autos['registration_year'].describe()

count    48565.000000
mean      2004.755421
std         88.643887
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

Cars released in 2004-2005 seem to be more. There are only outliers in our data. The values of 1000 and 9999 are outliers. Let's take a closer look at the registration_year column to improve the quality of the data.

In [30]:
autos['registration_year'].value_counts().sort_index().head(60)

1000      1
1001      1
1111      1
1800      2
1910      5
1927      1
1929      1
1931      1
1934      2
1937      4
1938      1
1939      1
1941      2
1943      1
1948      1
1950      3
1951      2
1952      1
1953      1
1954      2
1955      2
1956      4
1957      2
1958      4
1959      6
1960     23
1961      6
1962      4
1963      8
1964     12
1965     17
1966     22
1967     26
1968     26
1969     19
1970     38
1971     26
1972     33
1973     23
1974     24
1975     18
1976     21
1977     22
1978     44
1979     34
1980     85
1981     28
1982     41
1983     51
1984     51
1985     95
1986     72
1987     72
1988    135
1989    174
1990    347
1991    339
1992    370
1993    425
1994    629
Name: registration_year, dtype: int64

In [31]:
autos['registration_year'].value_counts().sort_index().tail(35)

1995    1227
1996    1373
1997    1951
1998    2363
1999    2897
2000    3156
2001    2636
2002    2486
2003    2699
2004    2703
2005    2936
2006    2670
2007    2277
2008    2215
2009    2085
2010    1589
2011    1623
2012    1310
2013     803
2014     663
2015     392
2016    1220
2017    1392
2018     470
2019       2
2800       1
4100       1
4500       1
4800       1
5000       4
5911       1
6200       1
8888       1
9000       1
9999       3
Name: registration_year, dtype: int64

Since the last date of creation of the advertisements was 2016, the ones after 2016 are outliner data. Since they will be very old in vehicles before 1900, we can accept them as outliner data.

In [32]:
autos = autos[(autos['registration_year'] > 1900) & (autos['registration_year'] <= 2016)]

In [33]:
autos['registration_year'].value_counts(normalize=True,ascending=True).head(60)

1952    0.000021
1931    0.000021
1927    0.000021
1948    0.000021
1938    0.000021
1939    0.000021
1929    0.000021
1943    0.000021
1953    0.000021
1955    0.000043
1957    0.000043
1934    0.000043
1951    0.000043
1941    0.000043
1954    0.000043
1950    0.000064
1962    0.000086
1937    0.000086
1958    0.000086
1956    0.000086
1910    0.000107
1961    0.000129
1959    0.000129
1963    0.000171
1964    0.000257
1965    0.000364
1975    0.000386
1969    0.000407
1976    0.000450
1966    0.000471
1977    0.000471
1973    0.000493
1960    0.000493
1974    0.000514
1971    0.000557
1968    0.000557
1967    0.000557
1981    0.000600
1972    0.000707
1979    0.000728
1970    0.000814
1982    0.000878
1978    0.000943
1984    0.001093
1983    0.001093
1986    0.001542
1987    0.001542
1980    0.001821
1985    0.002035
1988    0.002892
1989    0.003727
1991    0.007262
1990    0.007433
1992    0.007926
2015    0.008397
1993    0.009104
1994    0.013474
2014    0.014203
2013    0.0172

In [34]:
autos['registration_year'].value_counts(normalize=True,ascending=True).tail(18)

1995    0.026285
2012    0.028063
1996    0.029412
2010    0.034040
2011    0.034768
1997    0.041794
2009    0.044665
2008    0.047450
2007    0.048778
1998    0.050620
2002    0.053255
2001    0.056468
2006    0.057197
2003    0.057818
2004    0.057904
1999    0.062060
2005    0.062895
2000    0.067608
Name: registration_year, dtype: float64

The most advertised model is 2000 model vehicles. The least ones are the vehicles of 1952. The number of vehicles on the sales site between 1994 and 2016 is higher than in other years. One reason could be because they were scrapped as the vehicle's age would cause more problems. 

In [68]:
tryy_dict = {}
for tryy in autos['brand']:
    if tryy in tryy_dict:
        tryy_dict[tryy] += 1
    else:
        tryy_dict[tryy] = 1

In [70]:
selected_brands = []
for k,v in tryy_dict.items():
    if v / autos['brand'].value_counts().sum() > 5/100:
        selected_brands.append(k)        

In [71]:
selected_brands

['bmw', 'volkswagen', 'ford', 'mercedes_benz', 'audi', 'opel']

We determined the brands with the highest number of vehicles by creating a frequency table. By navigating this table, we selected the car brands that are more than 5% of the total rate.

In [80]:
mean_price = {}
for brands in selected_brands:
    means = autos[autos['brand'] == brands]['price'].mean()
    mean_price[brands] = round(means,2)

In [81]:
mean_price

{'bmw': 8332.82,
 'volkswagen': 5402.41,
 'ford': 3749.47,
 'mercedes_benz': 8628.45,
 'audi': 9336.69,
 'opel': 2975.24}

Audi, BMW and Mercedes Benz are more expensive,
 Ford and Opel are less expensive,
 Volkswagen is in between.

In [116]:
mean_price_series = pd.Series(mean_price)

In [117]:
mean_price_series

bmw              8332.82
volkswagen       5402.41
ford             3749.47
mercedes_benz    8628.45
audi             9336.69
opel             2975.24
dtype: float64

In [89]:
mean_km = {}
for brands in selected_brands:
    km = autos[autos['brand'] == brands]['odometer_km'].mean()
    mean_km[brands] = round(km,2)

In [90]:
mean_km

{'bmw': 132572.51,
 'volkswagen': 128707.16,
 'ford': 124266.01,
 'mercedes_benz': 130788.36,
 'audi': 129157.39,
 'opel': 129310.04}

In [118]:
mean_km_series = pd.Series(mean_km)

In [119]:
mean_km_series

bmw              132572.51
volkswagen       128707.16
ford             124266.01
mercedes_benz    130788.36
audi             129157.39
opel             129310.04
dtype: float64

In [140]:
df = pd.DataFrame(mean_price_series,columns=['mean_price'])

In [141]:
df

mean_price
bmw               8332.82
volkswagen        5402.41
ford              3749.47
mercedes_benz     8628.45
audi              9336.69
opel              2975.24

In [144]:
df['mean_km'] = mean_km_series

In [145]:
df

mean_price    mean_km
bmw               8332.82  132572.51
volkswagen        5402.41  128707.16
ford              3749.47  124266.01
mercedes_benz     8628.45  130788.36
audi              9336.69  129157.39
opel              2975.24  129310.04

There is no big difference in the average of vehicle kilometers. Therefore, we can say that the km has little or no effect on the price. The mileage difference does not seem much between the ones with low prices, even compared to the vehicles with a large price difference, the kms of those with low prices are also observed to be low. In our dataset, the brand stands out more.